packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,706 @@
+
+### Audit details
+
+To have better attribution in your audits you can use the `details` section in `AuditOutputs`.
+This helps to make the plugin results more actionable and valuable for the user.
+
+We will extend the `fileSizeAuditOutput` with `details` show which files exceed the defined budget.
+
+**file-size plugin runnerFunction**
+
+```typescript
+// file-size.plugin.ts
+// ...
+import { basename } from 'path';
+import { formatBytes, toUnixPath } from '@code-pushup/utils';
+import { AuditOutput } from './plugin-process-output';
+
+async function runnerFunction(options: Options): Promise<AuditOutputs> {
+  // ...
+
+  // assert file size information with budget
+  const issues = data.map(({ file, size }) => assertFileSizeInfo(file, size, options.budget));
+
+  // add details if issues given
+  if (issues.length) {
+    fileSizeAuditOutput = {
+      ...fileSizeAuditOutput,
+      details: {
+        issues,
+      },
+    };
+  }
+
+  return [fileSizeAuditOutput];
+}
+
+// assert file size info with budget
+export function assertFileSize(file: string, size: number, budget?: number): Issue {
+  const auditOutputBase = {
+    source: {
+      // format path to be in scope of the repository
+      file: toUnixPath(file, { toRelative: true }),
+    },
+  } satisfies AuditOutput['source'];
+
+  if (budget && budget < size) {
+    return {
+      ...auditOutputBase,
+      severity: 'error',
+      message: `File ${basename(filePath)} is ${formatBytes(size - budget)} too big. ( budget: ${formatBytes(budget)})`,
+    };
+  }
+
+  return {
+    ...auditOutputBase,
+    severity: 'info',
+    message: `File ${basename(filePath)} OK`,
+  };
+}
+```
+
+Test the output by running `npx code-pushup collect --format=md`.
+The CLI argument `--format=md` will create an additional file containing our created detail information from above.
+You should see a new file `report.md` created in the folder `.code-pushup` in your current working directory.
+The `report.md` file should contain a similar content like the following:
+
+<details>
+<summary> <b>report.md created by the CLI</b> (collapsed for brevity) </summary>
+
+```md
+<!-- ... -->
+
+### File Size Audit (File Size)
+
+<details>
+  <summary>🟨 <b>2 files</b> (score: 50)</summary>
+  <h4>Issues</h4>
+  <table>
+    <tr>
+      <th>Severity</th>
+      <th>Message</th>
+      <th>Source file</th>
+      <th>Line(s)</th>
+    </tr>
+    <tr>
+      <td>🚨 <i>error</i></td>
+      <td>File file-a.js is 17.31 kB bytes too big. (budget: 41.02 kB)</td>
+      <td><code>src/file-a.js</code></td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>ℹ️ <i>info</i></td>
+      <td>File file-b.js OK</td>
+      <td><code>src/file-b.js</code></td>
+      <td></td>
+    </tr>
+  </table>
+</details>
+```
+
+</details>
+
+## Plugins and categories
+
+In this chapter we will see how plugin results contribute to the category scoring.
+
+**basic categories setup**
```

Comment:
I would add some minimal information about weights (perhaps the group could have a higher weight?).

The second paragraph is taken from L43 in the file size plugin documentation - I think it fits better in this overview README rather than a specific example plugin documentation. What do you think?

```suggestion
**basic categories setup**
In this example we create a Performance category which contains one audit and one group.

Assign weights based on what influence each audit and group should have on the overall category score (assign weight 0 to only include it for extra info, without influencing the category score).
```